### PR TITLE
Improve performance of addSourceMap

### PR DESCRIPTION
**Pull request by devongovett**
This makes `addSourceMap` a lot faster by making it consume the source map it is adding rather than cloning all of the mappings. It improves overall perf on the esbuild benchmark by ~500ms on my machine, and cuts packaging time in half.

This should be safe for our current usage because we don't use the source map objects after adding them to another map. I also removed the column offsetting because it required us to re-insert all mappings into the BTreeMap which was expensive, and we don't currently use this feature either.

Also enabled jemalloc on macOS, which improves perf on all benchmarks by ~20%.

### CODE REVIEW

**Comment by devongovett** [comment]
Not sure why the benchmark action is still comparing against the rust rewrite commit vs the current master btw?

**Comment by mischnic** [comment]
Should we set a flag on the sourcemap object or something like that in case you try to use the consumed sourcemap for something else?

**Comment by devongovett** [comment]
Yeah maybe, but sounds like a future thing to me. We only use this method 4 places in one file at the moment so I'm not too worried.

File: test/append.test.js [hunk 1]
```diff
@@ -258,16 +258,88 @@ describe('SourceMap - Append Mappings', () => {
...
-    map.addBuffer(buffer, 10, 46);
-    assert.deepEqual(map.getMap(), expectedResultTwo);
+    let map2 = new SourceMap(map.toBuffer());
+    map.addSourceMap(map2, 10);
+    assert.deepEqual(map.getMap(), expectedResultOne);
+    // map2 is consumed
+    assert.equal(map2.getMappings().length, 0);
+  });
+
+  it('Append source map with line offset that overlap original lines', () => {
```
Comment (DeMoorJasper) [inline test/append.test.js:275]: This is another case where this new implementation fails.

I guess we have to choose between accuracy and speed here.

Comment (devongovett) [reply test/append.test.js:275]: Hmm, when would two source maps overlap?

Comment (DeMoorJasper) [reply test/append.test.js:275]: Yeah probably never, as it would probably produce an invalid sourcemap regardless if they overlap

**Review by DeMoorJasper — APPROVED** [review]

**Pull request merged**